tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+            ],
+        }
+        role_res = self.iam_client.create_role(
+            RoleName=iam_role_name,
+            AssumeRolePolicyDocument=json.dumps(trust_relationship_policy_ec2),
+            Description="Role for create custom KMS key",
+        )
+        policy_arn = self._create_iam_policies(iam_policy_name, scheduler)
+        # attach ParallelClusterPolicy to the role
+        logging.info("Attaching iam policy to the role {0}...".format(iam_role_name))
+        policy_attach_res = self.iam_client.attach_role_policy(RoleName=iam_role_name, PolicyArn=policy_arn)
+        time.sleep(20)
+        logging.info("Iam role is ready: {0}".format(iam_role_name))
+        return iam_role_name
+
+    def _create_iam_policies(self, iam_policy_name, scheduler):
+        # create the iam policy
+        # for different scheduler, attach different instance policy
+        logging.info("Creating iam policy {0} for iam role...".format(iam_policy_name))
+        file_loader = FileSystemLoader(pkg_resources.resource_filename(__name__, "/../../resources"))
+        env = Environment(loader=file_loader, trim_blocks=True, lstrip_blocks=True)
+        policy_filename = (
+            "batch_instance_policy.json" if scheduler == "awsbatch" else "traditional_instance_policy.json"
+        )
+        parallel_cluster_instance_policy = env.get_template(policy_filename).render(
+            partition="*", region="*", account_id=self.account_id, cluster_bucket_name="parallelcluster-*"
+        )
+
+        policy_res = self.iam_client.create_policy(
+            PolicyName=iam_policy_name, PolicyDocument=parallel_cluster_instance_policy
+        )
+        policy_arn = policy_res["Policy"]["Arn"]
+        return policy_arn
+
+    def _create_kms_key(self, scheduler):
+        # create KMS key
+        self.kms_client = boto3.client("kms")
+        if scheduler == "awsbatch":
+            key_alias = "alias/Integration_test_KMS_key_batch_{0}".format(self.region)
+        else:
+            key_alias = "alias/Integration3_test_KMS_key_{0}".format(self.region)
+        # If the key already existed, use the existing key
+        for alias in self.kms_client.list_aliases().get("Aliases"):
+            if alias.get("AliasName") == key_alias:
+                kms_key_id = alias.get("TargetKeyId")
+                logging.info("Use existing KMS key {0}".format(kms_key_id))
+                return kms_key_id
+
+        # if the key doesn't existed in the account, create a new key
+        logging.info("Creating KMS key...")
+        response = self.kms_client.create_key(
+            Description="create kms key",
+            KeyUsage="ENCRYPT_DECRYPT",
+            Origin="AWS_KMS",
+            BypassPolicyLockoutSafetyCheck=False,
+        )
+        kms_key_id = response["KeyMetadata"]["KeyId"]
+
+        # create KMS key policy
+        key_policy = {
```

Comment:
Minor: It might be nice to have larger json like this in a separate file so the code is less cluttered

Reply:
Good idea! Done